acceptance/launcher_test.go

```diff
@@ -42,217 +42,292 @@ func TestLauncher(t *testing.T) {
 }
 
 func testLauncher(t *testing.T, when spec.G, it spec.S) {
-	when("there is no CMD provided", func() {
-		when("CNB_PROCESS_TYPE is NOT set", func() {
-			it("web is the default process-type", func() {
-				cmd := exec.Command("docker", "run", "--rm", launchImage)
+	when("Platform API >= 0.4", func() {
+		when("entrypoint is a process", func() {
+			it("launches that process", func() {
+				entrypoint := "/cnb/process/web"
+				if runtime.GOOS == "windows" {
+					entrypoint = `c:\cnb\process\web.exe`
+				}
+				cmd := exec.Command("docker", "run", "--rm",
+					"--entrypoint="+entrypoint,
+					"--env=CNB_PLATFORM_API=0.4",
+					launchImage)
 				assertOutput(t, cmd, "Executing web process-type")
 			})
-		})
 
-		when("CNB_PROCESS_TYPE is set", func() {
-			it("should run the specified CNB_PROCESS_TYPE", func() {
-				cmd := exec.Command("docker", "run", "--rm", "--env", "CNB_PROCESS_TYPE=direct-process", launchImage)
+			it("appends any args to the process args", func() {
+				entrypoint := "/cnb/process/web"
 				if runtime.GOOS == "windows" {
-					assertOutput(t, cmd, "Usage: ping")
+					entrypoint = `c:\cnb\process\web`
+				}
+				cmd := exec.Command("docker", "run", "--rm",
+					"--entrypoint="+entrypoint,
+					"--env=CNB_PLATFORM_API=0.4",
+					launchImage, "with user provided args")
+				if runtime.GOOS == "windows" {
+					assertOutput(t, cmd, `Executing web process-type "with user provided args"`)
 				} else {
-					assertOutput(t, cmd, "Executing direct-process process-type")
+					assertOutput(t, cmd, "Executing web process-type with user provided args")
 				}
 			})
 		})
-	})
 
-	when("process-type provided in CMD", func() {
-		it("launches that process-type", func() {
-			cmd := exec.Command("docker", "run", "--rm", launchImage, "direct-process")
-			expected := "Executing direct-process process-type"
-			if runtime.GOOS == "windows" {
-				expected = "Usage: ping"
-			}
-			assertOutput(t, cmd, expected)
-		})
+		when("entrypoint is a not a process", func() {
+			it("builds a process from the arguments", func() {
+				cmd := exec.Command("docker", "run", "--rm",
+					"--entrypoint=/cnb/lifecycle/launcher",
+					"--env=CNB_PLATFORM_API=0.4",
+					launchImage, "--", "env")
```

Comment:
To avoid testing the full combinatorial explosion of APIs/args/direct (we would have a lot of tests) I was going to let the platform 0.3 test handle that, knowing that we should be in the same code path at that point.